pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -2336,375 +2543,414 @@ func TestDeleteNoDaemonPod(t *testing.T) {
-			newPod("non-matching-orphan-1-", "node-1", simpleDaemonSetLabel2, nil),
-			newPod("matching-owned-by-other-0-", "node-0", simpleDaemonSetLabel, ds2),
-		}
-		for _, pod := range ignoredPods {
-			manager.podStore.Add(pod)
-		}
+			// These pods should be ignored.
+			ignoredPods := []*v1.Pod{
+				newPod("non-matching-owned-0-", "node-0", simpleDaemonSetLabel2, ds),
+				newPod("non-matching-orphan-1-", "node-1", simpleDaemonSetLabel2, nil),
+				newPod("matching-owned-by-other-0-", "node-0", simpleDaemonSetLabel, ds2),
+			}
+			for _, pod := range ignoredPods {
+				manager.podStore.Add(pod)
+			}
 
-		nodesToDaemonPods, err := manager.getNodesToDaemonPods(ds)
-		if err != nil {
-			t.Fatalf("getNodesToDaemonPods() error: %v", err)
-		}
-		gotPods := map[string]bool{}
-		for node, pods := range nodesToDaemonPods {
-			for _, pod := range pods {
-				if pod.Spec.NodeName != node {
-					t.Errorf("pod %v grouped into %v but belongs in %v", pod.Name, node, pod.Spec.NodeName)
+			nodesToDaemonPods, err := manager.getNodesToDaemonPods(ds)
+			if err != nil {
+				t.Fatalf("getNodesToDaemonPods() error: %v", err)
+			}
+			gotPods := map[string]bool{}
+			for node, pods := range nodesToDaemonPods {
+				for _, pod := range pods {
+					if pod.Spec.NodeName != node {
+						t.Errorf("pod %v grouped into %v but belongs in %v", pod.Name, node, pod.Spec.NodeName)
+					}
+					gotPods[pod.Name] = true
 				}
-				gotPods[pod.Name] = true
 			}
-		}
-		for _, pod := range wantedPods {
-			if !gotPods[pod.Name] {
-				t.Errorf("expected pod %v but didn't get it", pod.Name)
+			for _, pod := range wantedPods {
+				if !gotPods[pod.Name] {
+					t.Errorf("expected pod %v but didn't get it", pod.Name)
+				}
+				delete(gotPods, pod.Name)
+			}
+			for podName := range gotPods {
+				t.Errorf("unexpected pod %v was returned", podName)
 			}
-			delete(gotPods, pod.Name)
-		}
-		for podName := range gotPods {
-			t.Errorf("unexpected pod %v was returned", podName)
 		}
 	}
 }
 
 func TestAddNode(t *testing.T) {
-	manager, _, _, err := newTestController()
-	if err != nil {
-		t.Fatalf("error creating DaemonSets controller: %v", err)
-	}
-	node1 := newNode("node1", nil)
-	ds := newDaemonSet("ds")
-	ds.Spec.Template.Spec.NodeSelector = simpleNodeLabel
-	manager.dsStore.Add(ds)
-
-	manager.addNode(node1)
-	if got, want := manager.queue.Len(), 0; got != want {
-		t.Fatalf("queue.Len() = %v, want %v", got, want)
-	}
-
-	node2 := newNode("node2", simpleNodeLabel)
-	manager.addNode(node2)
-	if got, want := manager.queue.Len(), 1; got != want {
-		t.Fatalf("queue.Len() = %v, want %v", got, want)
-	}
-	key, done := manager.queue.Get()
-	if key == nil || done {
-		t.Fatalf("failed to enqueue controller for node %v", node2.Name)
-	}
-}
-
-func TestAddPod(t *testing.T) {
-	for _, strategy := range updateStrategies() {
+	for _, f := range []bool{true, false} {
+		setFeatureGate(t, features.ScheduleDaemonSetPods, f)
```

Comment:
if we are setting a global value we should set it back to what it was before running this subtest (applies globally)

Reply:
So, if we are running the tests when the global value is enabled and disabled, I think it doesn't matter.

Reply:
I though some of the tests depend on the value being set initially, if one of the test changes that it has side effects. Why would you do the rollback in those other tests?

Reply:
You are right but say if one test has modified the flag it responsible for unsetting it. For example, `TestSimpleDaemonSetScheduleDaemonSetPodsLaunchesPods`, sets and unsets it.

Reply:
I think, in all the tests, we check the value(I believe you are calling this as default value). We set it or unset it explicitly. I don't mind adding rollback logic for every function.

Reply:
I guess we should either do the rollback in all the places or in none. Given it doesn't have a side effect in the unit test I won't block the PR.
/hold cancel
/lgtm